src/librustdoc/html/static/themes/dark.css

```diff
@@ -0,0 +1,382 @@
+.content .highlighted.trait { background-color: #013191; }
+.content .highlighted.mod { background-color: #803a1b; }
+.content .highlighted.externcrate { background-color: #afc6e4; }
+.content .highlighted.enum { background-color: #5b4e68; }
+.content .highlighted.struct { background-color: #194e9f; }
+.content .highlighted.fn,
+.content .highlighted.method,
+.content .highlighted.tymethod { background-color: #4950ed; }
+.content .highlighted.type { background-color: #38902c; }
+.content .highlighted.foreigntype { background-color: #f5c4ff; }
+.content .highlighted.macro { background-color: #8ce488; }
+.content .highlighted.constant,
+.content .highlighted.static { background-color: #c3e0ff; }
+.content .highlighted.primitive { background-color: #9aecff; }
+
+.content span.enum, .content a.enum, .block a.current.enum { color: #508157; }
+.content span.struct, .content a.struct, .block a.current.struct { color: #df3600; }
+.content span.type, .content a.type, .block a.current.type { color: #ba5d00; }
+.content span.foreigntype, .content a.foreigntype, .block a.current.foreigntype { color: #cd00e2; }
+.content span.macro, .content a.macro, .block a.current.macro { color: #068000; }
+.content span.union, .content a.union, .block a.current.union { color: #767b27; }
+.content span.constant, .content a.constant, .block a.current.constant,
+.content span.static, .content a.static, .block a.current.static { color: #546e8a; }
+.content span.primitive, .content a.primitive, .block a.current.primitive { color: #2c8093; }
+.content span.externcrate,
+.content span.mod, .content a.mod, .block a.current.mod { color: #967F00; }
+.content span.trait, .content a.trait, .block a.current.trait { color: #7c5af3; }
+.content span.fn, .content a.fn, .block a.current.fn,
+.content span.method, .content a.method, .block a.current.method,
+.content span.tymethod, .content a.tymethod, .block a.current.tymethod,
+.content .fnname{ color: #2BAB63; }
+
+pre.rust .comment { color: #4D4D4C; }
+pre.rust .doccomment { color: #8E908C; }
+
+nav {
+	border-bottom-color: #4e4e4e;
+}
+nav.main .current {
+	border-top-color: #eee;
+	border-bottom-color: #eee;
+}
+nav.main .separator {
+	border-color: #eee;
+}
+a {
+	color: #ddd;
+}
+
+.docblock a, .docblock-short a, .stability a {
+	color: #D2991D;
+}
+
+a.test-arrow {
+	color: #dedede;
+}
+
+.collapse-toggle {
+	color: #999;
+}
+
+.search-input {
+	color: #111;
+	box-shadow: 0 0 0 1px #000, 0 0 0 2px transparent;
+	background-color: #f0f0f0;
+}
+
+.search-input:focus {
+	border-color: #008dfd;
+}
+
+.stab.unstable { background: #FFF5D6; border-color: #FFC600; color: #404040; }
+.stab.deprecated { background: #F3DFFF; border-color: #7F0087;  color: #404040; }
+.stab.portability { background: #C4ECFF; border-color: #7BA5DB;  color: #404040; }
+
+.module-item .stab {
+	color: #ddd;
+}
+
+#help > div {
+	background: #e9e9e9;
+	border-color: #bfbfbf;
+}
+
+#help dt {
+	border-color: #bfbfbf;
+	background: #fff;
```

Comment:
With the standard `#dddddd` for the text, this gives 1.36:1. You can either change this background to `#4d4d4d` (same as the help div background), or add a `color: black;` to this block to make it the same as the light-theme ones.